Lib/asyncio/streams.py

```diff
@@ -952,17 +952,31 @@ def feed_data(self, data):
         If the data cannot be read because of over limit, a
         LimitOverrunError exception  will be raised, and the data
         will be left in the internal buffer, so it can be read again.
+
+        The ``separator`` may also be an iterable of separators. In this
+        case the return value will be the shortest possible that has any
+        separator as the suffix. For the purposes of LimitOverrunError,
+        the shortest possible separator is considered to be the one that
+        matched.
         """
-        seplen = len(separator)
-        if seplen == 0:
+        if isinstance(separator, bytes):
+            separator = [separator]
+        else:
+            # Makes sure shortest matches wins, and supports arbitrary iterables
+            separator = sorted(separator, key=lambda sep: len(sep))
+        if not separator:
+            raise ValueError('Separator list should contain at least one element')
```

Comment:
Since you support any iterable, maybe leave "list" out of it?

```suggestion
            raise ValueError('Separator should contain at least one element')
```